ntoskrnl/mm/ARM3/mminit.c

```diff
@@ -1819,12 +1819,12 @@ MiBuildPagedPool(VOID)
     //
     // Now check how many PTEs will be required for these many pages.
     //
-    Size = (Size + (1024 - 1)) / 1024;
+    Size = (Size + (PTE_PER_PAGE - 1)) / PTE_PER_PAGE;
 
     //
     // Recompute the page-aligned size of the paged pool, in bytes and pages.
     //
-    MmSizeOfPagedPoolInBytes = Size * PAGE_SIZE * 1024;
+    MmSizeOfPagedPoolInBytes = Size * PAGE_SIZE * PTE_PER_PAGE;
```

Comment:
This seems strange. Why does "size of paged pool in bytes" depend on how many PTEs fit in a page?

Reply:
The comments are misleading. Obviously this was written by Alex. And obviously he had no idea, what he was doing 😄 It's not calculating "how many PTEs will be required for these many pages", but how many PDEs are required. And somehow the logic is that paged pool is allocated at PDE granularity (there is an ASSERT below). Don't ask me why, ask Alex, maybe he knows why, but probably not, and probably newer Windows versions don't do this crazy stuff. Long story short: "It's what Windows does". I'll fix the comments.

- [ ] Fix comments.

Reply:
I fixed the comments as well as introduce additional variables other than "Size" to make stuff a little bit more readable,